### ISSUE TITLE
Error while re-deploying a stack that is ROLLBACK_COMPLETE state

### ISSUE THREAD

**Issue by nija-at** [aws/aws-cdk#8126]
This occurs once when a stack is in `ROLLBACK_COMPLETE` and a `cdk deploy` command is issued to attempt another deployment. **Retrying `cdk deploy` succeeds.**

As far as I can tell this is not a race condition since this occurs no matter how long you wait between the stack going into `ROLLBACK_COMPLETE` and the issuance of the deploy command.

### Reproduction Steps

1. Create a stack via `cdk deploy` that is going to throw an error from CloudFormation during deployment. The stack should end up in a `ROLLBACK_COMPLETE` state.
2. Re-issue the `cdk deploy` and the above error should show up.

A second retry of `cdk deploy` will succeed.

### Error Log

Error message when running `cdk deploy --app <path to js file>` -

```
integ-user-pool-idp-social: deploying...
integ-user-pool-idp-social: creating CloudFormation changeset...

 ❌  integ-user-pool-idp-social failed: Error [ValidationError]: Stack [integ-user-pool-idp-social] does not exist
    at Request.extractError (/Users/nija/.nvm/versions/node/v12.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/protocol/query.js:50:29)
    at Request.callListeners (/Users/nija/.nvm/versions/node/v12.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/Users/nija/.nvm/versions/node/v12.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/Users/nija/.nvm/versions/node/v12.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/Users/nija/.nvm/versions/node/v12.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/Users/nija/.nvm/versions/node/v12.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /Users/nija/.nvm/versions/node/v12.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/Users/nija/.nvm/versions/node/v12.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/Users/nija/.nvm/versions/node/v12.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:685:12)
    at Request.callListeners (/Users/nija/.nvm/versions/node/v12.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/sequential_executor.js:116:18) {
  code: 'ValidationError',
  time: 2020-05-21T10:45:53.471Z,
  requestId: '7b0e67c1-d3f0-4a60-bee0-722fec9fb71e',
  statusCode: 400,
  retryable: false,
  retryDelay: 290.0758012593673
}
Stack [integ-user-pool-idp-social] does not exist
```

### Environment

  - **CLI Version: 1.38.0**
  - **OS: OSX 10.14**

--- 

This is :bug: Bug Report

**Comment by shivlaks** [comment]
duplicate of #7340 and fixed by #8164 as of `1.42.0`

**Comment by eladb** [comment]
I am still running a similar issue (albeit with a different error message) in 1.45.0:

1. Deployed a stack
2. Tried to update, it failed and rolled back successfully. Now the status is ROLLBACK_COMPLETE
3. Try to update but the CLI fails with `The stack named ... is in a failed state`

From a quick inspection, it seems like the `isSuccess` property in `stack-status.ts` deems any stack with the word `ROLLBACK` in it's state to be unhealthy, which is not the case for ROLLBACK_COMPLETE.

I think the code in `stack-status.ts` has a few other issues. For example, the `isCreationFailure` property returns `true` only based on whether the status is ROLLBACK_*, and does not even check if the status was COMPLETE_FAILED at some point in the past.

This is a major issue - being unable to update a stack that is in ROLLBACK_COMPLETE is a huge blocker for users.
